src/Arcus.EventGrid.Testing/Infrastructure/Hosts/ServiceBus/ServiceBusEventConsumerHost.cs

```diff
@@ -0,0 +1,144 @@
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.EventGrid.Testing.Infrastructure.Hosts.ServiceBus
+{
+    /// <summary>
+    ///     Event consumer host for receiving Azure Event Grid events via Azure Logic Apps & Service Bus Topics
+    /// </summary>
+    public class ServiceBusEventConsumerHost : EventConsumerHost
+    {
+        private readonly SubscriptionClient _subscriptionClient;
+        private readonly ManagementClient _managementClient;
+
+        private static bool isHostShuttingDown;
+
+        private ServiceBusEventConsumerHost(string topicPath, string subscriptionName, SubscriptionClient subscriptionClient, ManagementClient managementClient, ILogger logger)
+            : base(logger)
+        {
+            Guard.NotNullOrWhitespace(topicPath, nameof(topicPath));
+            Guard.NotNullOrWhitespace(subscriptionName, nameof(subscriptionName));
+            Guard.NotNull(subscriptionClient, nameof(subscriptionClient));
+            Guard.NotNull(managementClient, nameof(managementClient));
+
+            TopicPath = topicPath;
+            SubscriptionName = subscriptionName;
+            _subscriptionClient = subscriptionClient;
+            _managementClient = managementClient;
+        }
+
+        /// <summary>
+        ///     Path of the topic relative to the namespace base address.
+        /// </summary>
+        public string TopicPath { get; }
+
+        /// <summary>
+        ///     Name of the subscription that was created
+        /// </summary>
+        public string SubscriptionName { get; }
+
+        /// <summary>
+        ///     Start receiving traffic
+        /// </summary>
+        /// <param name="topicPath">Path of the topic relative to the namespace base address</param>
+        /// <param name="serviceBusConnectionString">
+        ///     Connection string of the Azure Service Bus namespace to use to consume
+        ///     messages
+        /// </param>
+        /// <param name="logger">Logger to use for writing event information during the hybrid connection</param>
+        public static async Task<ServiceBusEventConsumerHost> Start(string topicPath, string serviceBusConnectionString, ILogger logger)
+        {
+            Guard.NotNullOrWhitespace(topicPath, nameof(topicPath));
+            Guard.NotNullOrWhitespace(serviceBusConnectionString, nameof(serviceBusConnectionString));
+            Guard.NotNull(logger, nameof(logger));
+
+            logger.LogInformation("Starting Service Bus event consumer host");
+
+            var managementClient = new ManagementClient(serviceBusConnectionString);
+            
+            var subscriptionName = $"Test-{Guid.NewGuid().ToString()}";
+            await CreateSubscriptionAsync(topicPath, managementClient, subscriptionName);
+            logger.LogInformation($"Created subscription '{subscriptionName}' on topic '{topicPath}'");
```

Comment:
I'd prefer to use the [structured logging](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.1#log-message-template) support of `ILogger`, for most log statements (not only this one)

```suggestion
            logger.LogInformation("Created subscription '{subscription}' on topic '{topic}'", subscriptionName, topicPath);
```